src/GraphQL/Types/Composite/AutoRegisteringInterfaceGraphType.cs

```diff
@@ -0,0 +1,154 @@
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace GraphQL.Types
+{
+    /// <summary>
+    /// Allows you to automatically register the necessary fields for the specified type.
+    /// Supports <see cref="DescriptionAttribute"/>, <see cref="ObsoleteAttribute"/>, <see cref="DefaultValueAttribute"/> and <see cref="RequiredAttribute"/>.
+    /// Also it can get descriptions for fields from the XML comments.
+    /// </summary>
+    public class AutoRegisteringInterfaceGraphType<TSourceType> : InterfaceGraphType<TSourceType>
```

Comment:
So `AutoRegisteringInterfaceGraphType` and `AutoRegisteringObjectGraphType` are precisely identical, except one derives from `InterfaceGraphType<T>` and the other from `ObjectGraphType<T>`.  Duplicated code is minimal, as most of it is contained within the `AutoRegisteringHelper` internal class.